### PR TITLE
front: fix pathfinding not relaunching if rollingstock is updated

### LINKED ISSUE #4787: Changing rolling stock does not relaunch pathfinding
### What happened?

In the train creation window, changing the rolling stock does not relaunch the pathfinding. This is problematic because the path can become invalid but the user doesn't know it. So it is possible to create a train without any error and it lead to this:

https://github.com/osrd-project/osrd/assets/86296016/ccb46e86-4a6d-44dd-827b-e1367ce07ce3

### What did you expect to happen?

Changing the rolling stock should either re-launch a pthfinding, or make the pathfinding tab mandatory again in order to force the user to re-launch it.

### How can we reproduce it (as minimally and precisely as possible)?

1. Go to any scenario
2. Reproduce the video above

### What operating system, browser and environment are you using?

- Browser: Chrome Version 115.0.5790.171
- OS: Windows 11
- Env: dev

### OSRD version (top right corner `Account` button > `Informations`)

Editoast 467ea7b Core 467ea7b API 467ea7b Front 467ea7b

### PR DESCRIPTION
**Pull request by SharglutDev**
close #4787

### CODE REVIEW

File: front/src/applications/operationalStudies/components/ManageTrainSchedule/SubmitConfAddTrainSchedule.tsx [hunk 1]
```diff
@@ -129,6 +129,14 @@ export default function SubmitConfAddTrainSchedule({ infraState, setIsWorking }:
         }
       }
     }
+    if (!osrdconf.simulationConf.pathfindingID) {
```
Comment (clarani) [inline front/src/applications/operationalStudies/components/ManageTrainSchedule/SubmitConfAddTrainSchedule.tsx]: To reach perfection haha

You can also move this test before the other one
```suggestion
    else if (!osrdconf.simulationConf.pathfindingID) {
```

File: front/src/applications/operationalStudies/components/ManageTrainSchedule/SubmitConfUpdateTrainSchedules.tsx [hunk 2]
```diff
@@ -78,6 +78,14 @@ export default function SubmitConfUpdateTrainSchedules({
         }
       }
     }
+    if (!osrdconf.simulationConf.pathfindingID) {
```
Comment (clarani) [inline front/src/applications/operationalStudies/components/ManageTrainSchedule/SubmitConfUpdateTrainSchedules.tsx]:
```suggestion
    else if (!osrdconf.simulationConf.pathfindingID) {
```

**Review by clarani — APPROVED** [review]
Nice job !! Only a suggestion :) tested in local :heavy_check_mark:

File: front/src/applications/operationalStudies/components/ManageTrainSchedule/SubmitConfAddTrainSchedule.tsx (hunk 1 above)
Comment (SharglutDev) [reply front/src/applications/operationalStudies/components/ManageTrainSchedule/SubmitConfAddTrainSchedule.tsx]: done :)

File: front/src/applications/operationalStudies/components/ManageTrainSchedule/SubmitConfUpdateTrainSchedules.tsx (hunk 2 above)
Comment (SharglutDev) [reply front/src/applications/operationalStudies/components/ManageTrainSchedule/SubmitConfUpdateTrainSchedules.tsx]: done :)

**Review by clarani — APPROVED** [review]
LGTM

**Pull request merged**